feat(link): new typography tokens, focus color and design, WHCM focus state (#3570)
adobe/spectrum-css: .changeset/tasty-eagles-draw.md

Reviewer: This will eventually get added automatically to the changelog in the component, so we'll want to think about how this reads in that context, could we make this a little more explanatory/documentation-y?

Some examples I saw that I liked were for [Breadcrumbs](https://github.com/adobe/spectrum-css/blob/spectrum-two/.changeset/fresh-crabs-deliver.md) and [Tooltip](https://github.com/adobe/spectrum-css/blob/spectrum-two/.changeset/light-years-speak.md), although certainly there are others in that `.changeset/` directory.

---
"@spectrum-css/link": major
---

> # Link S2 migration

Reviewer 2: ```suggestion
### Link S2 migration
```

Super minor knit but the only first-level heading in the changelogs is the top-level one. All the change sets should be h3s or lower

> The link component is now updated with S2 specs, colors, and typography.There is a new inline variant which uses the `.spectrum-Link--inline` class and the default variant is the standalone variant.

Reviewer 2: ```suggestion
The link component is updated with S2 specifications, colors, and typography. There is a new inline variant which uses the `.spectrum-Link--inline` modifier class and the default is the standalone variant.
```

A couple minor wording updates, can you say more about the "standalone" variant. Will most readers of the changelog know what that means or is there a way to describe it that might be more clear? Or can we link to the design documentation?

> **Notes**

Reviewer 2: ```suggestion
#### Note
```

The linter typically suggests using headings over bold text for a stronger navigability when rendered.

> - Quiet styling does not apply to the inline variant for accessiblity reasons.

Reviewer 2: ```suggestion
- Quiet styling does not apply to the inline variant for accessiblity reasons.
```

Any chance we can expand on those reasons a bit just to make sure we have them documented here for the changelog?

- `--mod-spectrum-link-font-weight` can be used to adjust inline variant to match surrounding text. Apply font weights to this mod `inherit` does not apply.

> ## New tokens

Reviewer 2: Are these new custom properties on the component-level or are these tokens from the token system that we're now using in the component styles?

Author: These are tokens from the token system now being used in the component.
